packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1576,7 +1577,7 @@ export const Dropdown: ComponentWithAs<'div', DropdownProps> &
                     )}
                 </div>
                 {showClearIndicator
-                  ? Box.create(clearIndicator, {
+                  ? Box.create(isEmpty(clearIndicator) ? <CloseIcon outline /> : clearIndicator, {
```

Comment:
Wouldn't be better to add it as `defaultProp` so we could avoid the `isEmpty` check?

Reply:
clearIndicator can hold text too, so we cannot escape the check somewhere in the code. what good would defaultProps do? running the check outside component?

Reply:
Would avoid check at all, there wouldn't be problem with text... See `Input` as example where we have:

```typescript
Input.defaultProps = {
  ...
  successIndicator: <PresenceAvailableIcon />,
};
```

Reply:
@assuncaocharles yeah.. seems a bit trolling but:
   ```
 const computeIcon = (): ShorthandValue<BoxProps> => {
      if (clearable && (value as string)?.length !== 0) {
        return {};
      }
      if (requiredAndSuccessful) {
        return successIndicator;
      }
      if (error) {
        return errorIndicator;
      }
      return icon || null;
    };
```
and later
```
{createShorthand(composeOptions.slots.icon, computeIcon(), {
```

Reply:
Input chooses between different icons too, see line 269, computeIcon()

Reply:
It would be avoiding `isEmpty` check at all, I just feel like more code to read and somehow unnecessary complexity

Reply:
I agree with @assuncaocharles  here, `computeIcon` has a different purpose, to choose between multiple icons. In this case the `empty` check can be avoided just by setting the default props which is correct behaviour

Reply:
@ling1726 @assuncaocharles Got it. Fixed.

Reply:
@ling1726 @assuncaocharles

Reply:
Thanks @TanelVari